apps/web/utils/actions/middleware.ts

```
@@ -1,17 +1,62 @@
+type EnsureObject<T> = T extends object ? T : never;
+
+/**
+ * Wraps an action in instrumentation to capture errors and send them to Sentry
+ * We also make sure to always return an object, so that if the client doesn't get a response,
+ * it's because there was an error, and likely it was a timeout because otherwise it would receive the response.
+ * We also handle thrown errors and convert them to an object with an error message.
+ * Actions are expected to return { error: string } objects when they fail, but if an error is thrown, we handle it here.
+ * NOTE: future updates: we can do more when errors are thrown like we do with `withError` middleware for API routes.
+ */
+export function withActionInstrumentation<
+  Args extends any[],
```

Biome

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)